### ISSUE TITLE
Uncaught Error: Repository has been destroyed

### ISSUE THREAD

**Issue by witeshadow** [jacogr/atom-git-control#26]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.179.0
**System**: Microsoft Windows 8.1
**Thrown From**: [git-control](https://github.com/jacogr/atom-git-control) package, v0.1.8

### Stack Trace

Uncaught Error: Repository has been destroyed

```
At C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\git-repository.js:382

Error: Repository has been destroyed
  at GitRepository.module.exports.GitRepository.getRepo (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\git-repository.js:382:15)
  at GitRepository.module.exports.GitRepository.getShortHead (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\git-repository.js:164:19)
  at Object.module.exports.getLocalBranch (C:\Users\Chris\.atom\packages\git-control\lib\git.coffee:106:17)
  at GitControlView.module.exports.GitControlView.loadBranches (C:\Users\Chris\.atom\packages\git-control\lib\git-control-view.coffee:82:27)
  at GitControlView.module.exports.GitControlView.update (C:\Users\Chris\.atom\packages\git-control\lib\git-control-view.coffee:54:6)
  at Object.module.exports.GitControl.updateViews (C:\Users\Chris\.atom\packages\git-control\lib\git-control.coffee:44:9)
  at atom-workspace.<anonymous> (C:\Users\Chris\.atom\packages\git-control\lib\git-control.coffee:17:43)
  at atom-workspace.handler (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\space-pen-extensions.js:110:34)
  at atom-workspace.jQuery.event.dispatch (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\node_modules\space-pen\vendor\jquery.js:4681:9)
  at atom-workspace.elemData.handle (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\node_modules\space-pen\vendor\jquery.js:4359:46)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\command-registry.js:243:29)
  at CommandRegistry.handleCommandEvent (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\command-registry.js:3:61)
  at CommandRegistry.module.exports.CommandRegistry.dispatch (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\command-registry.js:156:19)
  at PaneContainerView.jQuery.fn.trigger (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\space-pen-extensions.js:64:23)
  at PaneContainerView.module.exports.PaneContainerView.onActivePaneItemChanged (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\pane-container-view.js:53:19)
  at C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\pane-container-view.js:3:61
  at Emitter.module.exports.Emitter.emit (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\node_modules\event-kit\lib\emitter.js:82:11)
  at C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\pane-container.js:347:34
  at Emitter.module.exports.Emitter.emit (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\node_modules\event-kit\lib\emitter.js:82:11)
  at Pane.module.exports.Pane.setActiveItem (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\pane.js:264:22)
  at Pane.module.exports.Pane.activateItem (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\pane.js:328:21)
  at Object.module.exports.GitControl.toggleView (C:\Users\Chris\.atom\packages\git-control\lib\git-control.coffee:34:12)
  at atom-workspace.<anonymous> (C:\Users\Chris\.atom\packages\git-control\lib\git-control.coffee:16:48)
  at atom-workspace.handler (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\space-pen-extensions.js:115:32)
  at atom-workspace.jQuery.event.dispatch (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\node_modules\space-pen\vendor\jquery.js:4681:9)
  at atom-workspace.elemData.handle (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\node_modules\space-pen\vendor\jquery.js:4359:46)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\command-registry.js:243:29)
  at CommandRegistry.handleCommandEvent (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\command-registry.js:3:61)
  at CommandRegistry.module.exports.CommandRegistry.dispatch (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\command-registry.js:156:19)
  at EventEmitter.<anonymous> (C:\Users\Chris\AppData\Local\atom\app-0.179.0\resources\app\src\window-event-handler.js:65:30)
  at EventEmitter.emit (events.js:116:17)

```

### Commands

```
  2x -5:22.7 git-projects:toggle (div.settings-view.pane-item)
     -4:47.5 core:confirm (atom-text-editor.editor.mini)
  2x -4:16.1 git-control:toggle (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -1:20.5 git-plus:commit (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
  2x -1:18.0 git-plus:add-all (atom-text-editor.editor)
     -1:07.3 git-plus:commit (atom-text-editor.editor)
     -0:37.5 git-control:toggle (atom-text-editor.editor)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "koala",
      "timekeeper",
      "tualo-git-context",
      "SFTP-deployment",
      "Remote-FTP",
      "remote-sync",
      "recent-projects"
    ],
    "projectHome": "C:\\Users\\Chris\\Documents\\GitHub\\"
  }
}
```

### Installed Packages

```coffee
# User
autoclose-html, v0.15.0
build, v0.23.4
color-picker, v1.4.4
event-watch, v0.5.1
git-control, v0.1.8
git-log, v0.2.0
git-plus, v4.4.2
git-projects, v1.7.1
git-tab-status, v1.5.3
html-helper, v0.2.3
html-id-class-snippets, v1.4.1
html-img, v0.3.2
linter-pylint, v0.2.0
project-manager, v1.15.5
project-ring, v0.19.3
remote-edit, v1.6.4
script, v2.16.0
tree-view-git-projects, v0.2.1
wakatime, v4.0.0

# Dev
No dev packages
```

**Comment by hellerbarde** [comment]
This happens when you close all tabs (including git-control) and remove all project folders **after you successfully had the git-control tab open in a repo** and then try to open git-control.

**Comment by brylie** [comment]
I have this issue on Ubuntu 14.10. To resolve, I had to close Atom (1.x) and re-open.

**Comment by camlafit** [comment]
[Enter steps to reproduce below:]

1. Move between projects 
2. Open git control pane

**Atom Version**: 1.3.2
**System**: Debian GNU/Linux
**Thrown From**: [git-control](https://github.com/jacogr/atom-git-control) package, v0.4.0


### Stack Trace

Uncaught Error: Repository has been destroyed

```
At /usr/share/atom/resources/app.asar/src/git-repository.js:358

Error: Repository has been destroyed
    at GitRepository.module.exports.GitRepository.getRepo (/usr/share/atom/resources/app.asar/src/git-repository.js:358:15)
    at GitRepository.module.exports.GitRepository.getShortHead (/usr/share/atom/resources/app.asar/src/git-repository.js:165:19)
    at Object.module.exports.getLocalBranch (/home/km/.atom/packages/git-control/lib/git.coffee:115:17)
    at GitControlView.module.exports.GitControlView.loadBranches (/home/km/.atom/packages/git-control/lib/git-control-view.coffee:97:29)
    at GitControlView.module.exports.GitControlView.update (/home/km/.atom/packages/git-control/lib/git-control-view.coffee:67:8)
    at Object.module.exports.GitControl.updateViews (/home/km/.atom/packages/git-control/lib/git-control.coffee:44:9)
    at /home/km/.atom/packages/git-control/lib/git-control.coffee:17:57
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at /usr/share/atom/resources/app.asar/src/pane-container.js:333:27
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at Pane.module.exports.Pane.setActiveItem (/usr/share/atom/resources/app.asar/src/pane.js:248:22)
    at Pane.module.exports.Pane.activateItem (/usr/share/atom/resources/app.asar/src/pane.js:316:21)
    at Object.module.exports.GitControl.toggleView (/home/km/.atom/packages/git-control/lib/git-control.coffee:34:12)
    at atom-workspace.<anonymous> (/home/km/.atom/packages/git-control/lib/git-control.coffee:16:57)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/usr/share/atom/resources/app.asar/src/command-registry.js:260:29)
    at /usr/share/atom/resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:536:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:359:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/usr/share/atom/resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/usr/share/atom/resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -8:54.4.0 project-viewer:toggle (input.hidden-input)
     -0:00.7.0 git-control:toggle (input.hidden-input)
```

### Config

```json
{
  "core": {
    "themes": [
      "adwaita-pro-ui",
      "atom-dark-syntax"
    ],
    "audioBeep": false,
    "openEmptyEditorOnStart": false,
    "disabledPackages": [
      "atomatigit",
      "atom-php-navigation"
    ]
  }
}
```

### Installed Packages

```coffee
# User
adwaita-pro-ui, v2.1.3
auto-encoding, v0.5.2
color-picker, v2.0.14
git-control, v0.4.0
language-diff, v0.4.0
linter, v1.11.3
linter-php, v1.1.6
php-debug, v0.1.4
php-integrator-base, v0.3.1
php-integrator-navigation, v0.2.4
project-view, v0.4.1
project-viewer, v0.2.26

# Dev
No dev packages
```

**Comment by newelement** [comment]
[Enter steps to reproduce below:]

Move between projects
Open git control pane

**Atom Version**: 1.6.0-beta3
**System**: Mac OS X 10.11.3
**Thrown From**: [git-control](https://github.com/jacogr/atom-git-control) package, v0.4.0


### Stack Trace

Uncaught Error: Repository has been destroyed

```
At /Applications/Atom.app/Contents/Resources/app.asar/src/git-repository.js:369

Error: Repository has been destroyed
    at GitRepository.module.exports.GitRepository.getRepo (/Applications/Atom.app/Contents/Resources/app.asar/src/git-repository.js:369:15)
    at GitRepository.module.exports.GitRepository.getShortHead (/Applications/Atom.app/Contents/Resources/app.asar/src/git-repository.js:175:19)
    at Object.module.exports.getLocalBranch (/Users/donjones/.atom/packages/git-control/lib/git.coffee:115:17)
    at GitControlView.module.exports.GitControlView.loadBranches (/Users/donjones/.atom/packages/git-control/lib/git-control-view.coffee:97:29)
    at GitControlView.module.exports.GitControlView.update (/Users/donjones/.atom/packages/git-control/lib/git-control-view.coffee:67:8)
    at Object.module.exports.GitControl.updateViews (/Users/donjones/.atom/packages/git-control/lib/git-control.coffee:44:9)
    at /Users/donjones/.atom/packages/git-control/lib/git-control.coffee:17:57
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:346:27
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at Pane.module.exports.Pane.setActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:248:22)
    at Pane.module.exports.Pane.addItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:369:14)
    at Object.module.exports.GitControl.toggleView (/Users/donjones/.atom/packages/git-control/lib/git-control.coffee:32:19)
    at atom-workspace.<anonymous> (/Users/donjones/.atom/packages/git-control/lib/git-control.coffee:16:57)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:160:19)
    at module.exports._executeCallback (/Users/donjones/.atom/packages/tool-bar/lib/tool-bar-button-view.coffee:69:23)
    at space-pen-button.<anonymous> (/Users/donjones/.atom/packages/tool-bar/lib/tool-bar-button-view.coffee:29:7)
    at space-pen-button.jQuery.event.dispatch (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4435:9)
    at space-pen-button.elemData.handle (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/jquery/dist/jquery.js:4121:28)
```

### Commands

```
     -7:36.5.0 core:paste (atom-text-editor.editor.is-focused)
  7x -6:38.6.0 core:backspace (atom-text-editor.editor.is-focused)
     -6:33.6.0 core:copy (atom-text-editor.editor.is-focused)
     -6:31.9.0 docblockr:parse-enter (atom-text-editor.editor.is-focused)
     -6:31.9.0 editor:newline (atom-text-editor.editor.is-focused)
     -6:31.6.0 core:paste (atom-text-editor.editor.is-focused)
     -6:30.7.0 core:save (atom-text-editor.editor.is-focused)
     -6:27 git-control:toggle (atom-text-editor.editor)
  7x -4:51 core:move-left (atom-text-editor.editor.is-focused)
  2x -4:49.7.0 core:backspace (atom-text-editor.editor.is-focused)
     -4:46.4.0 core:move-right (atom-text-editor.editor.is-focused)
     -4:42.6.0 core:backspace (atom-text-editor.editor.is-focused)
     -4:42.4.0 core:save (atom-text-editor.editor.is-focused)
     -2:08.2.0 git-control:toggle (atom-pane.pane.active)
  2x -2:01.2.0 project-viewer:toggle (atom-pane.pane.active)
  9x -1:53.4.0 git-control:toggle (atom-pane.pane.active)
```

### Config

```json
{
  "core": {
    "themes": [
      "one-dark-ui",
      "an-old-hope-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
Remote-FTP, v0.7.19
an-old-hope-syntax, v0.5.0
angularjs, v0.3.4
atom-autocomplete-php, v0.18.7
atom-beautify, v0.28.22
color-picker, v2.1.1
docblockr, v0.7.3
emmet, v2.4.1
file-icons, v1.6.14
flex-tool-bar, v0.7.3
git-control, v0.4.0
indent-guide-improved, v1.4.5
linter, v1.11.3
linter-csslint, v1.3.1
linter-htmlhint, v1.0.2
linter-jshint, v1.3.1
linter-jsonlint, v1.1.5
linter-php, v1.1.8
linter-scss-lint, v2.5.1
minimap, v4.19.0
pigments, v0.24.0
project-viewer, v0.2.30
sync-settings, v0.6.0
terminal-plus, v0.14.5
tool-bar, v0.2.0
wordpress-api, v1.2.1

# Dev
No dev packages
```

**Comment by zzjin** [comment]
I have same issue here.
Find that when change projects paths using packages like (`project-viewer` or `project-manager` or `project-plus`)
They will call `atom.project.setPaths(paths)` when user change between open projects. But git-control not notice it. [Atom doc](https://atom.io/docs/api/v1.6.0/Project#instance-onDidChangePaths) says that you can watch this modifications.

**Comment by FezVrasta** [comment]
I have the same issue, seems still valid...

**Comment by bnb** [comment]
Just had the same issue. OS X Latest on a MacBook Pro Mid 2010.

**Comment by sparcut** [comment]
+1 Same issue. Windows 10. Atom 1.8.0

**Comment by spyro92** [comment]
the fix doesnt work

**Comment by spyro92** [comment]
i have fixed it in #191 

you can test by running 
```
apm install spyro92/atom-git-control
```

**Comment by micheleypf** [comment]
[Enter steps to reproduce:]

1. pulled from remote
2. From the file tree clicked "Remove project folder"

**Atom**: 1.17.2 x64
**Electron**: 1.3.15
**OS**: Ubuntu 16.04.2
**Thrown From**: [git-control](https://github.com/jacogr/atom-git-control) package 0.9.0


### Stack Trace

Uncaught Error: Repository has been destroyed

```
At /home/andrei/atom-1.11.1/out/app/node_modules/tree-view/lib/file.coffee:109748

Error: Repository has been destroyed
    at GitRepository.module.exports.GitRepository.getRepo (/opt/atom/resources/app/src/git-repository.js:370:1)
    at GitRepository.module.exports.GitRepository.getShortHead (/opt/atom/resources/app/src/git-repository.js:177:1)
    at Object.getLocalBranch (/packages/git-control/lib/git.coffee:129:17)
    at GitControlView.module.exports.GitControlView.loadBranches (/packages/git-control/lib/git-control-view.coffee:125:29)
    at GitControlView.module.exports.GitControlView.update (/packages/git-control/lib/git-control-view.coffee:94:8)
    at Object.updateViews (/packages/git-control/lib/git-control.coffee:51:9)
    at /packages/git-control/lib/git-control.coffee:19:57
    at Function.module.exports.Emitter.simpleDispatch (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at Workspace.didActivatePaneContainer (/opt/atom/resources/app/src/workspace.js:474:1)
    at WorkspaceCenter.paneContainer.onDidActivatePane (/opt/atom/resources/app/src/workspace-center.js:11:1)
    at Function.module.exports.Emitter.simpleDispatch (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at PaneContainer.didActivatePane (/opt/atom/resources/app/src/pane-container.js:260:1)
    at Pane.module.exports.Pane.activate (/opt/atom/resources/app/src/pane.js:875:1)
    at WorkspaceCenter.activate (/opt/atom/resources/app/src/workspace-center.js:34:1)
    at Workspace.didHideDock (/opt/atom/resources/app/src/workspace.js:550:1)
    at Dock.setState (/opt/atom/resources/app/src/dock.js:154:1)
    at Dock.handleDidRemovePaneItem (/opt/atom/resources/app/src/dock.js:242:1)
    at Function.module.exports.Emitter.simpleDispatch (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at Pane.module.exports.Pane.removeItem (/opt/atom/resources/app/src/pane.js:613:1)
    at TreeViewPackage.createOrDestroyTreeViewIfNeeded (/opt/atom/resources/app/node_modules/tree-view/lib/tree-view-package.js:85:1)
    at Function.module.exports.Emitter.simpleDispatch (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:25:1)
    at Emitter.module.exports.Emitter.emit (/opt/atom/resources/app/node_modules/event-kit/lib/emitter.js:141:1)
    at Project.module.exports.Project.removePath (/opt/atom/resources/app/src/project.js:322:1)
    at TreeView.module.exports.TreeView.removeProjectFolder (/opt/atom/resources/app/node_modules/tree-view/lib/tree-view.js:1252:1)
    at HTMLDivElement.atom.commands.add.tree-view:remove-project-folder (/opt/atom/resources/app/node_modules/tree-view/lib/tree-view.js:385:1)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/opt/atom/resources/app/src/command-registry.js:265:1)
    at CommandRegistry.handleCommandEvent (/opt/atom/resources/app/src/command-registry.js:3:1)
    at CommandRegistry.module.exports.CommandRegistry.dispatch (/opt/atom/resources/app/src/command-registry.js:166:1)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchContextMenuCommand (/opt/atom/resources/app/src/atom-environment.js:1306:1)
    at EventEmitter.outerCallback (/opt/atom/resources/app/src/application-delegate.js:347:1)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

### Commands

```
     -2:24.7.0 tree-view:show (atom-workspace.workspace.scrollbars-visible-always)
     -2:15.9.0 intentions:highlight (input.hidden-input)
     -2:13.8.0 git-control:toggle (input.hidden-input)
     -1:42.6.0 tree-view:remove-project-folder (span.name.icon.icon-repo)
```

### Non-Core Packages

```
atom-beautify 0.29.26 
busy-signal 1.4.1 
cobalt2-syntax 0.5.1 
file-icons 2.1.7 
git-control 0.9.0 
go-debug 1.4.2 
go-plus 5.5.5 
go-signature-statusbar 1.2.0 
hyperclick 0.0.40 
intentions 1.1.2 
linter 2.1.4 
linter-jshint 3.1.2 
linter-ui-default 1.6.0 
node-debugger 1.10.1 
platformio-ide-terminal 2.5.1 
```
